app/scripts/migrations/127.test.ts

```diff
@@ -0,0 +1,796 @@
+              nickname: 'Random Network',
+              ticker: 'FOO',
+              rpcUrl: 'https://localhost/rpc/2',
+              rpcPrefs: {
+                blockExplorerUrl: 'https://localhost/explorer',
+              },
+            },
+          },
+        },
+      },
+    };
+
+    const newState = await migrate(oldState);
+
+    const { networkConfigurationsByChainId } = newState.data
+      .NetworkController as {
+      networkConfigurationsByChainId: Record<
+        string,
+        {
+          defaultBlockExplorerUrlIndex: number;
+          blockExplorerUrls: string[];
+        }
+      >;
+    };
+
+    const networkConfiguration = networkConfigurationsByChainId[randomChainId];
+    expect(networkConfiguration.defaultBlockExplorerUrlIndex).toStrictEqual(0);
+    expect(networkConfiguration.blockExplorerUrls).toStrictEqual([
+      'https://localhost/explorer',
+    ]);
+  });
+
+  it('dedupes if there are duplicate rpc urls within a chain id', async () => {
+    for (const [url1, url2] of [
+      ['http://test.endpoint/bar', 'http://test.endpoint/bar'],
+      // Check case insensitivity (network controller requires this)
+      ['http://test.endpoint/bar', 'HTTP://TEST.ENDPOINT/bar'],
+    ]) {
+      const randomChainId = '0x123456';
+
+      const oldState = {
+        meta: { version: oldVersion },
+        data: {
+          NetworkController: {
+            selectedNetworkClientId: 'sepolia',
+            networkConfigurations: {
+              'network-id-1': {
+                id: 'network-id-1',
+                chainId: randomChainId,
+                nickname: 'Random Network',
+                ticker: 'FOO',
+                rpcUrl: url1,
+              },
+              'network-id-2': {
+                id: 'network-id-2',
+                chainId: randomChainId,
+                nickname: 'Random Network',
+                ticker: 'FOO',
+                rpcUrl: url2,
+              },
+            },
+          },
+        },
+      };
+
+      const newState = await migrate(oldState);
+
+      const { networkConfigurationsByChainId } = newState.data
+        .NetworkController as {
+        networkConfigurationsByChainId: Record<
+          string,
+          {
+            defaultRpcEndpointIndex: number;
+            rpcEndpoints: {
+              url: string;
+              type: string;
+              name: string;
+              networkClientId: string;
+            }[];
+          }
+        >;
+      };
+
+      const networkConfiguration =
+        networkConfigurationsByChainId[randomChainId];
+      expect(networkConfiguration.defaultRpcEndpointIndex).toStrictEqual(0);
+      expect(networkConfiguration.rpcEndpoints).toStrictEqual([
+        {
+          url: url1,
+          type: 'custom',
+          name: 'Random Network',
+          networkClientId: 'network-id-1',
+        },
+      ]);
+    }
+  });
+
+  it('dedupes if there are duplicate rpc urls across chain ids', async () => {
```

Comment:
is it possible for this case to exist in state? Or are we being extra safe and handling it just in case?

Reply:
It's definitely possible, although unlikely.  The network controller is stricter in its validation now.  It introduced a library uri-js to compare url equality, which handles things like different casing that the network form UI would not have prevented.